pkg/crd/kubermatic/v1/cluster.go

```diff
@@ -291,20 +291,20 @@ type AuditLoggingSettings struct {
 }
 
 type ComponentSettings struct {
-	Apiserver         APIServerSettings       `json:"apiserver"`
-	ControllerManager DeploymentSettings      `json:"controllerManager"`
-	Scheduler         DeploymentSettings      `json:"scheduler"`
-	Etcd              EtcdStatefulSetSettings `json:"etcd"`
-	Prometheus        StatefulSetSettings     `json:"prometheus"`
+	Apiserver         APIServerSettings   `json:"apiserver"`
+	ControllerManager ResourceSettings    `json:"controllerManager"`
+	Scheduler         ResourceSettings    `json:"scheduler"`
+	Etcd              ResourceSettings    `json:"etcd"`
+	Prometheus        StatefulSetSettings `json:"prometheus"`
 }
 
 type APIServerSettings struct {
-	DeploymentSettings `json:",inline"`
+	ResourceSettings `json:",inline"`
 
 	EndpointReconcilingDisabled *bool `json:"endpointReconcilingDisabled,omitempty"`
 }
 
-type DeploymentSettings struct {
+type ResourceSettings struct {
```

Comment:
The replica count is logically not really expected to be nested below `resources`, so I would tend to disagree with this renaming. Also, `DeploymentSettings` allows for easier extensibility.